build.sbt

```
@@ -31,48 +59,48 @@ val unnamedJavaOptions = List(
   "--add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED"
 )
 
-lazy val root = project
-  .in(file("."))
+lazy val root = (projectMatrix in file("."))
   .aggregate(encoders, udf, examples)
   .settings(publish / skip := true)
   .settings(publishSettings)
 
-lazy val encoders = project
-  .in(file("encoders"))
+lazy val encoders = sparkVersionMatrix(projectMatrix in file("encoders"))
```

I suggest you move the `projectMatrix` part to `sparkVersionMatrix` to avoid redudancy

```suggestion
lazy val encoders = sparkVersionMatrix(file("encoders"))
```